File: frontend/src/types/telemetry.ts
export interface TelemetryLog {
  id: string;
  timestamp: string;
  sessionId: string;
  endpoint: string;
  statusCode: number;
  ip: string;
  requestFrequency: number;
  anomalyFlag?: string;
}

export interface DecisionOption {
  id: string;
  label: string;
}

export interface DecisionPayload {
  decisionId: string;
  triggerType: string;
  currentRiskScore: number;
  prompt: string;
  options: DecisionOption[];
  hints: string[];
}

export type SocketMessage =
  | { type: "TELEMETRY"; data: TelemetryLog }
  | { type: "DECISION_REQUIRED"; data: DecisionPayload };

File: telemetry.py
import time

class TelemetryGenerator:
    @staticmethod
    def create_log(log_id: str, endpoint: str, status_code: int, ip: str, frequency: int, anomaly_flag: str):
        return {
            "type": "TELEMETRY",
            "data": {
                "id": log_id,
                "timestamp": time.strftime("%H:%M:%S"),
                "endpoint": endpoint,
                "statusCode": status_code,
                "ip": ip,
                "requestFrequency": frequency,
                "anomalyFlag": anomaly_flag
            }
        }